### PR TITLE
Add configurable search scope (word/translation) to word list

### PR DESCRIPTION
**Pull request by Vladyslav-Soldatenko**
## Description

The word list search field only matched the `word` field, so users couldn't find a card by its translation. This adds a tune icon to the search bar that opens a "Search in" dialog letting users choose Word, Translation, or both. At least one option must stay checked (Apply is disabled otherwise, with an inline error). The choice is persisted via DataStore and applied immediately to the active search query.

## Type of Change

- [x] New feature
- [ ] Bug fix
- [ ] Refactoring (no functional changes)
- [ ] Documentation update
- [ ] Infrastructure / CI/CD
- [ ] Performance improvement

## Changes Made

- Added `SearchScope` domain model (`matchWord`, `matchTranslation`, `isValid`).
- Added `WordListSearchPreferencesStore` (DataStore-backed, defaults to both `true`), following the existing `LanguagePreferencesStore` pattern — reuses `StudyPreferencesDataStore`'s underlying store, no new Hilt module needed.
- `WordListViewModel` now exposes `searchScope: StateFlow<SearchScope>` and `setSearchScope(...)`, guarding against ever persisting an all-false scope.
- Added `WordListSearchScopeDialog`: an `AlertDialog` titled "Search in" with Word/Translation checkboxes, Apply (disabled + error text when nothing is checked) and Cancel buttons. Dismissing (Cancel, back, tap-outside) discards unapplied changes.
- Added a trailing `Icons.Default.Tune` icon inside the existing search field that opens the dialog.
- Updated the word-list filter predicate to respect the active scope (word-only / translation-only / both, with no duplicate results when both fields match).
- Added `action_apply` and `word_list_search_scope_*` strings, translated into all 15 supported locales (reusing existing `add_word_label_word`/`add_word_label_translation` values where identical, to avoid `DuplicateStrings` lint errors).
- Exposed `WordListSearchPreferencesStore` via `PreferencesEntryPoint` so instrumented tests can reset/verify persisted scope directly.

## How to Test

1. Open the word list, add a couple of words where the query matches only the word or only the translation.
2. Tap the tune icon next to the search field — a "Search in" dialog appears with Word and Translation both checked.
3. Uncheck one option, confirm Apply stays enabled with one checked, then uncheck the other — Apply disables and an error appears. Re-check one to re-enable Apply.
4. Apply a word-only or translation-only scope and confirm the search results respect it; confirm Cancel/back/outside-tap discards the change.
5. Restart the app and confirm the applied scope persisted.

## Checklist

- [x] Code follows the project's style guidelines
- [x] Self-reviewed the code for obvious errors
- [x] Added or updated tests where applicable, including edge cases (unit tests for the new DataStore, ViewModel, and dialog/filter UI behavior, plus new instrumented e2e tests)
- [x] Existing tests pass locally (`ktlintCheck`, `detekt`, `lintDebug`, `testDebugUnitTest` all green; instrumented tests can't run in this environment — CI will exercise them)
- [ ] Updated documentation if needed (no user-facing docs exist for this screen)
- [x] No duplication introduced. I searched the existing code for similar behavior and extracted similar parts to be reused
- [ ] No new warnings or suppressions introduced — added `@Suppress("LargeClass")` to `WordListScreenTest` since detekt's LargeClass threshold was crossed by the added tests; this mirrors the existing precedent in `AddWordViewModelTest.kt` and `AppDatabaseMigrationTest.kt`. Flagging for maintainer approval per the checklist above.

## Related Issues

N/A

---
_Generated by [Claude Code](https://claude.ai/code/session_015XGRLhEFhas98XzK474dHc)_

### CODE REVIEW

File: app/src/test/java/com/procrastilearn/app/ui/screens/WordListScreenTest.kt [hunk 1]
```diff
@@ -37,6 +39,7 @@ import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
 
 @RunWith(RobolectricTestRunner::class)
+@Suppress("LargeClass")
```
Comment (Vladyslav-Soldatenko) [inline app/src/test/java/com/procrastilearn/app/ui/screens/WordListScreenTest.kt]: extract all test cases from this file related to search to separate file to avoid supression

Comment (Vladyslav-Soldatenko) [reply app/src/test/java/com/procrastilearn/app/ui/screens/WordListScreenTest.kt]: Done — moved all search-related tests (query filtering, tune icon, search-scope dialog) into a new `WordListSearchTest.kt`. `WordListScreenTest.kt` is back under the LargeClass threshold, so I dropped the `@Suppress`. Pushed in b9a5257.

---
_Generated by [Claude Code](https://claude.ai/code)_
